### PR TITLE
lua_api.txt clarifications regarding mapgen object index IDs and handles.

### PR DESCRIPTION
**Pull request by shacknetisp**
Currently there is no mention that these are different, so it took a while of trial and error before I discovered that `minetest.get_decoration_id`'s return was actually different from `minetest.register_decoration`'s return value.

### CODE REVIEW

**Comment by paramat** [comment]
:+1:

**Comment by SmallJoker** [comment]
`minetest.register_ore` and `minetest.register_biome` "suffer" from the same issue. Please refer to the mapgen object-specific getter functions for all of them.

**Comment by shacknetisp** [comment]
I've generalized this more to differentiate between biome/decoration/etc. index IDs and their object handles.

File: doc/lua_api.txt [hunk 1]
```diff
@@ -3750,20 +3750,20 @@ Call these functions only at load time!
       mapgens. See [Mapgen aliases] section above.
 * `minetest.register_alias_force(alias, original_name)`
 * `minetest.register_ore(ore definition)`
-    * Returns an integer uniquely identifying the registered ore on success.
+    * Returns an integer object handle uniquely identifying the registered ore on success.
     * The order of ore registrations determines the order of ore generation.
 * `minetest.register_biome(biome definition)`
-    * Returns an integer uniquely identifying the registered biome on success.
+    * Returns an integer object handle uniquely identifying the registered biome on success. To get the biome ID, use `minetest.get_biome_id`.
```
Comment (paramat) [inline doc/lua_api.txt]: Wrap line at 80 columns (with tab size 4)

**Comment by shacknetisp** [comment]
Outside the scope of this PR, but there are some tabulators here: https://github.com/minetest/minetest/blob/8efa1de8356630f2943e3d74e7287b5e976e8706/doc/lua_api.txt#L2062-L2067

**Comment by paramat** [comment]
:+1:

**Pull request merged**